### DISCUSSION TITLE
How to change namespace url of variable?

### DISCUSSION THREAD

**Discussion by VasilyBroze** [FreeOpcUa/opcua-asyncio#1355] [Q&A] [answered]
When I create variable, it's reserve id (i) for it, and then url for taking value looks like /example.freeopcua.io/*i. 
Can I use name of variable instead of this id? (like /example.freeopcua.io/Light1)
Can it be customized?

**Comment by schroeder-** [comment]
You can read the namespace array via `get_namespace_array `on the server or client.
Also in OPC UA you always have the namespace + the identifer. As Identifer you can also use a string.

```Python
idx = await server.get_namespace_index("/example.freeopcua.io")
ua.NodeId("Light1", idx)
```
or this
```Python
ua.NodeId("/example.freeopcua.io/Light1", idx)
```

**Reply by VasilyBroze** [reply]
Thanks for answer! 
I create variables like this:
idx = server.register_namespace(uri)
myobj = objects.add_object(idx, "MyObj")
myobj.add_variable(idx, "Light1", 0.0, ua.VariantType.Float)
and standard identifer is integer number like /example.freeopcua.io/1 but not /example.freeopcua.io/Light1 (log prints n=1, i=1)
as I understand it, the method that you described above changes the Node ID to string, but not the Variable ID, right?
Can I change identifer of variable?

**Reply by schroeder-** [reply, marked as answer]
You can pass a nodeid instead of the namespace index:
`myobj.add_variable(ua.NodeId("Light1", idx), "Light1", 0.0, ua.VariantType.Float)`
Or like this:
`myobj.add_variable(f"ns={idx};s=Light1", "Light1", 0.0, ua.VariantType.Float)`
or like this:
`myobj.add_variable("ns=1;s=Light1", "Light1", 0.0, ua.VariantType.Float)`

**Reply by VasilyBroze** [reply]
Thank you! It's works :)
